components/nodemanager-service/tests/project_update_test.go

```diff
@@ -0,0 +1,2231 @@
+package manager
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	iam_v2 "github.com/chef/automate/api/interservice/authz/v2"
+	"github.com/chef/automate/components/nodemanager-service/api/manager"
+	"github.com/chef/automate/components/nodemanager-service/api/nodes"
+	"github.com/chef/automate/components/nodemanager-service/config"
+	"github.com/chef/automate/components/nodemanager-service/pgdb"
+	"github.com/gofrs/uuid"
+	"github.com/golang/protobuf/ptypes"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestProjectUpdate(t *testing.T) {
+	ctx := context.Background()
+	db, err := createPGDB()
+	require.NoError(t, err)
+
+	timestamp, err := ptypes.TimestampProto(time.Now())
+	require.NoError(t, err)
+
+	cases := []struct {
+		description        string
+		projectsData       []*nodes.ProjectsData
+		projectRules       map[string]*iam_v2.ProjectRules
+		originalProjectIDs []string
+		expectedProjectIDs []string
+	}{
+		// Environment
+		{
+			description: "Environment - updating project",
+			projectsData: []*nodes.ProjectsData{
+				{Key: "environment", Values: []string{"env1"}},
+			},
+			projectRules: map[string]*iam_v2.ProjectRules{
+				"project3": {
```

Comment:
Suggest using a named constant like `targetProject` for this line and L57 to make the intent a bit clearer.